API.pm

```diff
@@ -158,6 +158,7 @@ sub cacheTrackMetadata {
 			artist => $entry->{artist},
 			artists => $entry->{artists},
 			album => $entry->{album}->{title},
+			album_details => $entry->{album},
```

Comment:
I'd prefer not to store all the album object, as this sometimes in turn would include the track list again. Just add `albumId`.

Reply:
Ok.  FYI the deezer plugin grabs the whole album information.

I have a problem with the name conventions here.  Is it `albumId` or `album_id`.  It is `replay_gain` three lines below, and `favorites_url` is a key for a item property elsewhere.    Esp. you translate the camel_case: `replay_gain => $entry->{replayGain}`, it's like the API (outside data) uses pascal but *we* translate that to underscores.

Reply:
I think `my $albumId = $track->remote ? $remoteMeta->{album_id} : undef;` is correct.  The *key*/mapping is underscore but the var name is pascal.

Reply:
It's possible Deezer is doing that. I'd still prefer it the other way around. It also depends on the data source: some services would include all track information with an album, some would provide all album information with each track, and some don't mix them at all. You can't always copy from one to another.

As for the variable naming convention: it's... confusing at best. There's none, really. I usually prefer camel case. But sometimes I'd also use the style that the previous developer had used, even if it was different.

Reply:
I indeed do that because it's a small amount of information and it allowed me to have the full album data in other places, which was convenient (I initially added just the album id and reverted that) but I understand that if the album object includes all tracks, you would not want that.